### ISSUE TITLE
Error message after saving branch [racing condition problem]

### ISSUE THREAD

**Issue by pameck** [rabblerouser/core#157]
### Error Description: 
After a branch is created, the following message is displayed:

<img width="1280" alt="screen shot 2017-04-12 at 9 15 13 am" src="https://cloud.githubusercontent.com/assets/3664374/24934754/e690bbb4-1f60-11e7-8c10-3ac3b23d246a.png">

In the dev console, there is this:

<img width="957" alt="screen shot 2017-04-12 at 9 15 41 am" src="https://cloud.githubusercontent.com/assets/3664374/24934753/e56c1936-1f60-11e7-8cdd-7ceef4a37af5.png">

### To replicate:
1. Go to the admin dashboard
2. Create a new branch
3. See error.

### Possible cause: 

When a new branch is created, there are two subsequent requests: 

 `GET /branches/{id}/groups`
 `GET /branches/{id}/members`

The problem is that we're adding the new branch to the core's store after it has gone through the event pipe, lambdas and back to core, but the requests from the UI get to the core before the new branch is added to the store which is why those two requests get a `404` error.

The following diagram is for reference about the branch creation flow: 

![branches-with-events 1](https://cloud.githubusercontent.com/assets/3664374/24935103/23ff0350-1f63-11e7-8873-8074dee03692.png)

**Comment by yearofthedan** [comment]
It's an interesting problem. A couple of options come to mind.
- The front end could use the information it has for the branch rather than making a request to the backend after saving. This would mean the FE should set the id in the initial create request for future use.
- We could wait a fixed time before requesting the branch, but that feels quite hacky.
- We could poll the backend for the data after we save and then enforce a timeout. This would be a little chattier.

**Comment by camjackson** [comment]
Wait, so, why is the frontend even making those requests? I don't think it's an ID generation issue, because the backend generates the ID before putting the event on the stream, and it will return that ID to the frontend synchronously.

From @pameck's screenshot, it seems the frontend is trying to fetch the new branch's groups and members, which is probably unnecessary. I think it's safe to assume that a brand new branch doesn't have any groups or members yet, so those fields can just be initialised to empty arrays, rather than fetching empty arrays from the backend.

So proposed solution: just don't make those requests 😁

**Comment by yearofthedan** [comment]
I havent looked too deep, but I think the issue is that the app selects the newly created branch which triggers a new API call for the branch details. So that would mean either not selecting, or creating some new flow.

**Comment by pameck** [comment]
I think that on top of the change in the UI, the API should return an empty array instead of a `404` when asking for a collection of things.

**Comment by yearofthedan** [comment]
But also you're right; ignore what I said about IDs. I had a blip and forgot that the create request returns one.

**Comment by yearofthedan** [comment]
Even if the branch can't be found?

**Comment by pameck** [comment]
@yearofthedan right! It's the branch that's missing not the groups or members, back to the `404` 😬
